src/components/inputlonganswer/InputLongAnswer.js

```diff
@@ -0,0 +1,70 @@
+import styled from "styled-components";
+
+import { colors } from "../colors/index";
+import { 
+    InputLongAnswerCSS,
+    InputCSS,
+    ErrorMessageCSS 
+} from "./InputLongAnswerCSS";
+
+const InputLongAnswerComponent = (props) => {
+    const getInputVariantStyles = (p) => {
+        const isFilledState = `
+        color: ${colors.emerald400};  
+        &:focus {
+            border-color: ${colors.emerald400};
+        }
+        `;
+        const errorState = `
+        border: 0.25rem solid ${colors.red500};
+        &:focus {
+            border: 0.25rem solid ${colors.red500};
+        }
+        `;
+
+        let styles = "";
+
+        if (p.isFilled) {
+            styles += isFilledState;
+        }
+
+        if (p.aboveCharLimit || p.noAnswer) {
+            styles += errorState;
+        }
+
+        return styles;
+    }
+
+    const InputLongAnswer = styled.div`
+        ${InputLongAnswerCSS}
+    `;
+
+    const Input = styled.textarea`
+        ${InputCSS}
+        ${(p) => getInputVariantStyles(p)}
+    `;
+
+    const ErrorMessage = styled.span`
+        ${ErrorMessageCSS}
+    `;
+
+    return (
+        <InputLongAnswer isFilled={props.isFilled} aboveCharLimit={props.aboveCharLimit} noAnswer={props.noAnswer}>
```

Comment:
i'm questioning whether we need all these props. ideally, the component would be smart enough to be able to tell that it should be in an error state by checking that the text value has gone over the character limit. one way you can do this is to use react states to track the text value and the useeffect hook to check if the text value has exceeded the character limit. 
this way the only prop you'll only need is for the error state when the input is empty 

`const [text, setText] = useState("");`

`  const changeHandler = (e) => {
    setText(e.target.value);
  };`

`  useEffect(() => {
    if (text.length > 650) {
      console.log("text value over 650");
      // use a react state here to manage error for text exceeding max character limit
    }
  });`

` <Input
value={text}
        onChange={changeHandler}
</Input>`